src/QsCompiler/Transformations/CallGraph.cs

```diff
@@ -0,0 +1,569 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.Quantum.QsCompiler.DataTypes;
+using Microsoft.Quantum.QsCompiler.SyntaxTokens;
+using Microsoft.Quantum.QsCompiler.SyntaxTree;
+
+
+// ToDo: Review access modifiers
+
+namespace Microsoft.Quantum.QsCompiler.Transformations.CallGraphNS
```

Comment:
Do you have any suggestions? Everything in this namespace is for the sake of the Call Graph class, but we can't call the namespace CallGraph because that is the name of the class. It causes lots of problems if the namespace shares a name with a contained type.

Reply:
How about Microsoft.Quantum.QsCompiler.DependencyAnalysis? There are more things down the line that we will need for dependency analysis.

Reply:
Without the "Transformations"? This namespace also covers the the call graph walker. Should that have a separate namespace from the types in this file?

Reply:
That might make extra sense if we are going to have the call graph be something that lives beyond a single rewrite step. If that's the case, it should not live in a "Transformations" ns. But that still leaves us with the namespace for the call graph walker.